src/docs/asciidoc/http-client.adoc

```diff
@@ -433,3 +433,106 @@ public class Application {
+
+== Metrics
+HTTP client supports built-in integration with https://micrometer.io/[`Micrometer`].
+It exposes all metrics with prefix `reactor.netty.http.client`.
+
+The following table provides information for the HTTP client metrics:
+
+[width="100%",options="header"]
+|=======
+| metric name | type | description
+| reactor.netty.http.client.data.received | DistributionSummary | Amount of the data that is received, in bytes
+| reactor.netty.http.client.data.sent | DistributionSummary | Amount of the data that is sent, in bytes
+| reactor.netty.http.client.errors | Counter | Number of the errors that are occurred
+| reactor.netty.http.client.tls.handshake.time | Timer | Time that is spent for TLS handshake
+| reactor.netty.http.client.connect.time | Timer | Time that is spent for connecting to the remote address
+| reactor.netty.http.client.address.resolver | Timer | Time that is spent for resolving the address
+| reactor.netty.http.client.data.received.time | Timer | Time that is spent in consuming incoming data
+| reactor.netty.http.client.data.sent.time | Timer | Time that is spent in sending outgoing data
+| reactor.netty.http.client.response.time | Timer | Total time for the request/response
+|=======
+
+These additional metrics also will be available:
+
+include::conn-provider-metrics.adoc[]
+
+include::alloc-metrics.adoc[]
+
+The following example enables that integration:
+
+====
+[source,java]
+----
+import io.micrometer.core.instrument.Metrics;
+import io.micrometer.core.instrument.config.MeterFilter;
+import reactor.netty.http.client.HttpClient;
+
+public class Application {
+
+    public static void main(String[] args) {
+        Metrics.globalRegistry <1>
+               .config()
+               .meterFilter(MeterFilter.maximumAllowableTags("reactor.netty.http.client", "URI", 100, MeterFilter.deny()));
+
+        HttpClient client =
+                HttpClient.create()
+                          .metrics(true, s -> {
+                              if (s.startsWith("/stream/")) { <2>
+                                  return "/stream/{n}";
+                              }
+                              else if (s.startsWith("/bytes/")) {
+                                  return "/bytes/{n}";
+                              }
+                              return s;
+                          }); <3>
+
+        client.get()
+              .uri("http://httpbin.org/stream/2")
+              .responseContent()
+              .blockLast();
+
+        client.get()
+              .uri("http://httpbin.org/bytes/1024")
+              .responseContent()
+              .blockLast();
+    }
+}
+----
+<1> Applies upper limit for the meters with `URI` tag
+<2> Templated URIs will be used as an URI tag value when possible
+<3> Enables the built-in integration with Micrometer
+====
+
+NOTE: In order to avoid an overhead of the enabled metrics it is important to convert the real URIs to templated URIs when possible.
```

Comment:
I would add more information about the "why" (convey that without conversion to template-like form, each distinct URI leads to the creation of a distinct tag, which takes a toll on memory footprint of metrics)

Reply:
the note below can reference this note to say putting a hard cap on the number of meters can at least help if one doesn't provide a uri `Function`

Reply:
done